test/boost/README.md

```diff
@@ -0,0 +1,80 @@
+# Scylla unit tests using C++ and the Boost test framework
+
+The source files in this directory are Scylla unit tests written in C++
+using the Boost.Test framework. These unit tests come in three flavors:
+
+1. Some simple tests that check stand-alone C++ functions or classes use
+   Boost's `BOOST_AUTO_TEST_CASE`.
+
+2. Some tests require Seastar features, and need to be declared with Seastar's
+   extensions to Boost.Test, namely `SEASTAR_TEST_CASE`.
+
+3. Even more elaborate tests require not just a functioning Seastar environment
+   but also a complete (or partial) Scylla environment. Those tests use the
+   `do_with_cql_env()` or `do_with_cql_env_thread()` function to set up a
+   mostly-functioning environment behaving like a single-node Scylla, in
+   which the test can run.
+
+While we have many tests of the third flavor, writing new tests of this
+type should be reserved to *white box* tests - tests where it is necessary
+to inspect or control Scylla internals that do not have user-facing APIs
+such as CQL. In contrast, black-box tests - tests that **can** be written
+only using user-facing APIs, should be written in one of newer test frameworks
+that we offer - such as test/cql-pytest or test/alternator (in Python, using
+the CQL or DynamoDB APIs respectively) or test/cql (using textual CQL
+commands), or - if more than one Scylla node is needed for a test - using
+the test/topology* framework.
```

Comment:
test/topology is planned to merge into cql-pytest, so I would not mention it here explicitly.

Reply:
This idea of combining cql-pytest with the much smaller and very different api'ed (and Scylla-only) topology tests does not make sense, let's discuss it separately before we ruin a good thing.